### PR TITLE
feat: create update schematic for cms component state

### LINKED ISSUE #6027: Schematic and deprecation of cms component state
This is a follow up deprecation ticket of work done in #4603 

To do:
- ~~Search for `TODO(issue:6027)` across the code and follow the instruction.~~ To be done in #6420
- Write an update schematic for an automatic upgrade to 2.0

### PR DESCRIPTION
**Pull request by znikola**
closes #6027

### CODE REVIEW

File: projects/schematics/src/migrations/2_0/validate.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export function validate(): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const angularVersion = getAngularVersion(tree);
+    validateAngularVersion(angularVersion);
+
+    return tree;
+  };
+}
+
+export function validateAngularVersion(angularVersion: string): void {
```
Comment (psmul) [inline projects/schematics/src/migrations/2_0/validate.ts]: Maybe we can go with something more generic and move it to shared/utils?

Comment (znikola) [reply projects/schematics/src/migrations/2_0/validate.ts]: Not sure if this should be a generic method. We could always do that of course, but I don't see it being use in multiple places for now, besides this validation. Do you have some specific use case on mind?

**Comment by znikola** [comment]
>     1. In file **update-cms-component-state_spec**, line 436 we have unit test with description 'ultimate test'. Can we make it more descriptive?

Sure, I renamed it to _all test cases in one file_.

>     2. When running **yarn build** in schematics source project we get an error about missing type for shelljs. Had to run command `npm install @types/shelljs`.

Good catch, I added it to the schematics' `package.json` and generated a new `yarn.lock`

File: projects/schematics/src/migrations/2_0/validate.ts (hunk 1 above)
Comment (psmul) [reply projects/schematics/src/migrations/2_0/validate.ts]: Tried to come up with some example, but maybe I overreacted with this one :) Let's leave it as is.

Comment (znikola) [reply projects/schematics/src/migrations/2_0/validate.ts]: It's a valid proposition :)

File: package.json [hunk 2]
```diff
@@ -158,6 +159,7 @@
     "sass": "^1.22.10",
     "semver": "^6.3.0",
     "semver-extra": "^3.0.0",
+    "shelljs": "^0.8.3",
```
Comment (dunqan) [inline package.json]: Can you help me understand why we need this dependency?

File: projects/cds/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@spartacus/cds",
-  "version": "1.0.0-next.0",
+  "version": "2.0.0",
```
Comment (dunqan) [inline projects/cds/package.json]: I believe we should set `2.0.0-next.0` for all versions, will it break the logic?
(we should be able to run `ng update @spartacus/schematics@next` until 2.0 becomes the final)

Comment (znikola) [reply projects/cds/package.json]: Done

File: package.json (hunk 2 above)
Comment (znikola) [reply package.json]: It's required for spec files, where we use it to switch directories.

**Pull request merged**